dpnp/dpnp_iface_statistics.py

```diff
@@ -352,69 +350,104 @@ def histogram(a, bins=10, range=None, density=None, weights=None):
-    if x1_desc:
-        # Negative values in 'shape' are not allowed in input array
-        # 306-322 check on negative and duplicate axis
-        isaxis = True
-        if axis is not None:
-            if dpnp.isscalar(axis):
-                if axis < 0:
-                    isaxis = False
-            else:
-                for val in axis:
-                    if val < 0:
-                        isaxis = False
-                        break
-                if isaxis:
-                    for i in range(len(axis)):
-                        for j in range(len(axis)):
-                            if i != j:
-                                if axis[i] == axis[j]:
-                                    isaxis = False
-                                    break
-
-        if not isaxis:
-            pass
-        elif out is not None:
-            pass
-        elif keepdims:
-            pass
-        elif initial is not None:
-            pass
-        elif where is not True:
-            pass
+    if initial is not None:
+        raise NotImplementedError(
+            "initial keyword arguemnts is only supported by its default value."
+        )
+    elif where is not True:
+        raise NotImplementedError(
+            "where keyword arguemnts is only supported by its default values."
+        )
+    else:
+        dpt_array = dpnp.get_usm_ndarray(a)
+        if dpt_array.size == 0:
+            # TODO: get rid of this if condition when dpctl supports it
+            for i in range(a.ndim):
+                if a.shape[i] == 0:
+                    if i not in axis:
+                        indices = [i for i in range(a.ndim) if i not in axis]
+                        res_shape = tuple([a.shape[i] for i in indices])
+                        result = dpnp.empty(res_shape, dtype=a.dtype)
+                    else:
+                        raise ValueError(
+                            "reduction does not support zero-size arrays"
+                        )
         else:
-            result_obj = dpnp_max(x1_desc, axis).get_pyobj()
-            result = dpnp.convert_single_elem_array_to_scalar(result_obj)
-
+            result = dpnp_array._create_from_usm_ndarray(
+                dpt.max(dpt_array, axis=axis, keepdims=keepdims)
+            )
+        if out is None:
             return result
-
-    return call_origin(numpy.max, x1, axis, out, keepdims, initial, where)
+        else:
+            if out.shape != result.shape:
+                raise ValueError(
+                    f"Output array of shape {result.shape} is needed, got {out.shape}."
+                )
+            elif out.dtype != result.dtype:
+                raise ValueError(
+                    f"Output array of type {result.dtype} is needed, got {out.dtype}."
+                )
+            elif not isinstance(out, dpnp_array):
+                if isinstance(out, dpt.usm_ndarray):
+                    out = dpnp.array(out)
```

Comment:
we should not create a memory copy of `out`, but only to wrap `usm_ndarray` into `dpnp array`:
```suggestion
                    out = dpnp_array._create_from_usm_ndarray(out)
```